src/Mod/Assembly/Gui/ViewProviderAssembly.cpp

```diff
@@ -112,6 +114,20 @@ QIcon ViewProviderAssembly::getIcon() const
     return Gui::BitmapFactory().pixmap("Geoassembly.svg");
 }
 
+void ViewProviderAssembly::setupContextMenu(QMenu* menu, QObject* receiver, const char* member)
+{
+    auto func = new Gui::ActionFunction(menu);
+
+    QAction* act = menu->addAction(QObject::tr("Active object"));
+    act->setCheckable(true);
+    act->setChecked(isActivePart());
+    func->trigger(act, [this]() {
+        this->doubleClicked();
+    });
+
+    ViewProviderDragger::setupContextMenu(menu, receiver, member);
```

Comment:
Did you intentionally skip over ViewProviderPart here? It makes the linter cranky.

Reply:
Yes exactly it's intended. `ViewProviderPart ` is implementing a QAction that is setting the part as active. We are overriding this QAction here.

Reply:
I have just added a `//NOLINT`